handler/handler.go

```diff
@@ -4,18 +4,63 @@ import (
 	"encoding/json"
 	"fmt"
 	"log"
+	"sync"
+	"time"
 
 	"github.com/nats-io/go-nats-streaming"
 	"github.com/openfaas/faas/gateway/queue"
 )
 
 // NatsQueue queue for work
 type NatsQueue struct {
-	nc        stan.Conn
-	ClientID  string
-	ClusterID string
-	NATSURL   string
-	Topic     string
+	nc                    stan.Conn
+	ncMutex               *sync.RWMutex
+	maxReconnect          int
+	delayBetweenReconnect time.Duration
+
+	ClientID              string
+	ClusterID             string
+	NATSURL               string
+	Topic                 string
+}
+
+func (q *NatsQueue) connect() error {
+	nc, err := stan.Connect(
+		q.ClusterID,
+		q.ClientID,
+		stan.NatsURL(q.NATSURL),
+		stan.SetConnectionLostHandler(func(conn stan.Conn, err error) {
+			log.Printf("Disconnected from %s\n", q.NATSURL)
+
+			q.reconnect(0)
```

Comment:
@kozlovic should we try to reconnect here? You mentioned adding this handler in the original older PR:

https://github.com/openfaas/nats-queue-worker/pull/33#discussion_r205115177

Reply:
Here is a good place to reconnect. This callback will be invoked in a go routine by the NATS Streaming library when the connection is lost for good.